### PR TITLE
Improve notebook support

### PR DESCRIPTION
**Pull request by grantjllnl**
Implemented many changes to better support Jupyter access to Wintap data via DuckDB.
Parameterized queries using new annotations.py per dataset to avoid hard-coded values.

### CODE REVIEW

File: rawutil.py [hunk 1]
```diff
@@ -168,7 +179,13 @@ def generate_view_sql(event_map):
         create or replace view {event_type} as
         select * from parquet_scan('{pathspec}',hive_partitioning=1)
         """
+        if start!=None and end!=None:
+            stmt += f'where dayPK between {start} and {end}'
+        if start!=None and end==None:
+            stmt += f'where dayPK = {start}'
         stmts.append(view_sql)
+        logging.debug(f"View for {event_type} using {pathspec}")
```
Comment (lindsey-w) [inline rawutil.py:187]: for consistency, should this be single quote?

**Review by lindsey-w — APPROVED** [review]
🥳  Just had some minor comments on format

File: rawutil.py [hunk 2]
```diff
@@ -113,10 +115,19 @@ def get_globs_for(dataset, daypk):
...
+                try:
+                    table = pq.read_table(file)
+                    if table.num_rows == 0:
+                        logging.info(f"{file} is empty, deleting.")
+                        os.remove(file)
+                except ArrowInvalid:
+                    # Move invalid files out of the way
+                    # Move to dataset/invalid/path
+                    logging.error(f'Invalid parquet: {file}')
+                    os.rename(file,file+'.invalid')
```
[the remark this answers is not in the document]
Comment (grantjllnl) [reply rawutil.py]: Thanks for spotting that. I went thru the other py files and fixed a few others like that.

File: rawutil.py (hunk 1 above)
Comment (grantjllnl) [reply rawutil.py]: Yes, single quote should be that default. I fixed all that I could. There some cases where there are embedded single quotes, so doubles are used.

**Pull request merged**